docker-app/qfieldcloud/core/utils2/storage.py

```diff
@@ -26,6 +31,69 @@ def staticfile_prefix(project: "Project", filename: str) -> str:  # noqa: F821
     return ""
 
 
+def file_response(
+    request: HttpRequest,
+    key: str,
+    presigned: bool = False,
+    expires: int = 60,
+    version: str = None,
+    as_attachment: bool = False,
+) -> HttpResponseBase:
+    url = ""
+    filename = PurePath(key).name
+    extra_params = {}
+
+    if version is not None:
+        extra_params["VersionId"] = version
+
+    # check if we are in NGINX proxy
+    if request.META.get("HTTP_HOST", "").split(":")[-1] == request.META.get(
+        "WEB_HTTPS_PORT"
+    ):
+        if presigned:
+            if as_attachment:
+                extra_params["ResponseContentType"] = "application/force-download"
+                extra_params[
+                    "ResponseContentDisposition"
+                ] = f'attachment;filename="{filename}"'
+
+            url = qfieldcloud.core.utils.get_s3_client().generate_presigned_url(
+                "get_object",
+                Params={
+                    **extra_params,
+                    "Key": key,
+                    "Bucket": qfieldcloud.core.utils.get_s3_bucket().name,
+                },
+                ExpiresIn=expires,
+                HttpMethod="GET",
+            )
+        else:
+            url = qfieldcloud.core.utils.get_s3_object_url(key)
+
+        # Let's NGINX handle the redirect to the storage and streaming the file contents back to the client
+        response = HttpResponse()
+        response["X-Accel-Redirect"] = "/storage-download/"
+        response["redirect_uri"] = url
+
+        return response
+    elif os.environ.get("DEBUG", 1):
```

Comment:
better to use `settings.DEBUG`, or maybe even don't add this alternative behavior so we keep dev as close as possible to prod ?

(here I tend to always run the full stack, and think it's acceptable to have download not work if accessing to django directly)

Reply:
Ah, now I saw I made default to 1, which was dangerous. Thanks for highlighting this!